build/constraints.js

```diff
@@ -849,7 +928,189 @@ function splitNMCmin(sTree,pTree)
+ * the helper function minimizes code repitition by taking only one tree and
+ * being called for both ptree and stree.
+ */
+function equalStrengthHelper(tree, cat, edgeName){
+  var boundaries = 0;
+  recursiveStrengthHelper(tree, cat, boundaries);
+  //recursiveStrengthHelper only goes through the left/right branches of tree
+  //arguments given unique names to avoid scope problems
+  function recursiveStrengthHelper(partree, category, boundaries){
+    // argument boundaries keeps track of the number of edges a terminal is on
+    var edgeIndex; // the index of the l/r edge
+    //left = [0]
+    if (edgeName == "left"){
+      edgeIndex = 0;
+    }
+    //right = [length -1]
+    if (edgeName == "right" && partree.children && partree.children.length){
+      edgeIndex = (partree.children.length - 1);
+    }
+    //increment boundaries when partree is of the correct category
+    if (partree.cat === category){
+      boundaries ++;
+    }
+    //if partree is terminal and partree.edges is not already defined
+    //edges must not already be defined because function is called recursively
+    if (!partree.children && !partree.edges){
+      partree.edges = boundaries; //assign partree.edges the value of boundaries
+    }
+    //recursively call inner function for the child on the l/r edge
+    if (partree.children && partree.children.length){
+      recursiveStrengthHelper(partree.children[edgeIndex], category, boundaries);
+    }
+  }
+  //recursively call outer function for every node
+  if (tree.children && tree.children.length){
+    for (var i = 0; i < tree.children.length; i ++){
+      equalStrengthHelper(tree.children[i], cat, edgeName);
+    }
+  }
+}
+
+/* A function to return the paired category as defined in categoryPairings.
+ * categoryPairings only returns prosodic categories given a syntactic category.
+ * reversableCatPairings also returns a syntactic category given a prosodic
+ * category.
+ * Maybe this should move to prosodicHierarchy.js?
+*/
+function reversableCatPairings(cat){
+  if (categoryPairings[cat]){
+    return categoryPairings[cat]; //just the same as calling categoryPairings
+  }
+  else {
+    //get the property names of categoryPairings
+    var props = Object.getOwnPropertyNames(categoryPairings);
+    var propFound = false; //true when the category is paired
+    for (var i = 0; i < props.length; i ++){
+      if (categoryPairings[props[i]] == cat){
+        propFound = true;
+        if (props[i] == "clause"){
+          // rn categoryPairings has a property "clause" which maps to i
+          return "cp"; // "cp" also maps to i, I think we want "cp"
+        }
+        //props[i] is the property that maps to cat
+        return props[i];
+      }
+    }
+    // if no matching category is found, return a costom error.
+    if (!propFound){
+      var err = new Error("" + cat + " is not a category defined in categoryPairings (see main/prosodicHierarchy.js)");
```

Comment:
Throw this error instead of returning it:
throw(new Error("blah blah"));

Reply:
changed in latest commit